enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -199,214 +153,90 @@ rules:
-the underlying, associated Pod references CSI volumes using `Shares`
-- or the cluster admin modifies the user's object(s) so they get their `Shares`
-
-Second, a few considerations around how we update objects, so they "get" `Shares` associated with them:
-
-- Pod objects themselves, or any API object that exposes the Pod's `volumes` array as part of providing an API that is
-converted by a controller into a Pod, can directly add a volume of the CSI type with the correct metadata in the volumeAttributes, a la
-
-```yaml
-csi:
-  driver: projected-resources.storage.openshift.io
-    volumeAttributes:
-      share: the-projected-resource
-
-```
-
-- with it as part of a Pod for example:
-
- ```yaml
-apiVersion: v1
-kind: Pod
-metadata:
-  name: some-pod
-spec:
-  containers:
-    ...
-     volumeMounts:
-        - name: shared-item-name
-          mountPath: /path/to/shared-item-data
-  volumes:
-      - name: entitlement
-        csi:
-          driver: projectedresources.storage.openshift.io
-          volumeAttributes:
-             share: the-projected-resource
-
-```
-
-- API objects which do not expose the resulting Pod's volume array will need to have an annotation set on it which triggers
-an admission plugin to mutate the Pod and add the CSI volume.
-
-On that new admission plugin introduced and its validation/mutation of Pods:
-- Pods with this annotation are mutated, adding the CSI driver volume specification,
-- if the Pod `ServiceAccount` is allowed sufficient
-access to the requested `ProjectedResource` (i.e. a SAR check for the SA getting the requested `ProjectedResource` passes)
-- if a Pod already happens to have the CSI Volume specified with reference to a `ProjectedResource`, again the plugin will see if the
-associated `ServiceAccount` for the Pod has sufficient access to the `ProjectedResource`
-- if sufficient access does not exist, the admission plugin marks the Pod as invalid/forbidden.
-- if sufficient access does exists, the admission plugin adds a CSI volume into the Pod's `volume` array, to match the
-example noted above.
-
+The cluster admin is then responsible for assigning this cluster role to appropriate serivce accounts that consume the share (either directly or via role aggregation).
+
+Pods then mount the shared resource via a `csi` volume.
+All CSI volumes must specify a `driver` that is installed on the cluster.
+The pod can also specify `volumeAttributes`, a free-form object whose schema is specific to each CSI driver.
+
+The initial API for the driver's `volumeAttributes` will consist of the `share` field, whose value is a reference to the cluster `Share` object.
+
+  ```yaml
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    name: some-pod
+  spec:
+    containers:
+      ...
+      volumeMounts:
+      - name: cluster-share
+        mountPath: /path/to/shared-item-data
+    volumes:
+    - name: cluster-share
+      csi:
+        driver: projectedresources.storage.openshift.io
```

Comment:
at least at the moment, the driver name is `csi-driver-projected-resource.openshift.io`  .... per https://github.com/openshift/csi-driver-projected-resource/blob/master/examples/csi-app.yaml#L18

But of course if we envision changing the name as part of our path to the CVO, that is good as well.